### PR TITLE
feat: KPI-232 save user preferences in report

### PR DESCRIPTION
**Pull request by dsgarcia8**
## Jira Link

[Go to Jira](https://kpinetwork.atlassian.net/browse/KPI-232)

## Types of changes

- [ ] Hotfix (Bug detected)
- [x] Feature (Issue from an Active Sprint)

## Checklist:

- [ ] Unit Test.
- [x] Coverage.

### CODE REVIEW

File: src/components/Header.jsx [hunk 1]
```diff
@@ -88,8 +88,10 @@ const PageMenu = ({ onClose }) => {
 
 const ProfileMenu = ({ signOut, visible, onProfileClose }) => {
   const onSignOut = () => {
-    localStorage.removeItem('year')
-    localStorage.removeItem('filters')
+    const localItems = ['year', 'filters', 'calendarYear', 'metrics', 'metric', 'firstMetric', 'secondMetric']
+    localItems.map((item) =>
+      localStorage.removeItem(item)
```
Comment (kellycastrof) [inline src/components/Header.jsx:93]: Are there more items in local storage, why not use the clear function? localStorage.clear()

File: src/utils/useLocalStorage.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-export const SaveLocalStorage = (key, value) => {
-  localStorage.setItem(key, value)
+export const setLocalStorage = (key, value) => {
+  return localStorage.setItem(key, JSON.stringify(value))
 }
 
-export const SaveJsonLocalStorage = (key, value) => {
-  localStorage.setItem(key, JSON.stringify(value))
+export const getLocalStorage = (key) => {
```
Comment (kellycastrof) [inline src/utils/useLocalStorage.js:5]: I think getFromLocalStorage is a better name

File: src/utils/useLocalStorage.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
-export const SaveLocalStorage = (key, value) => {
-  localStorage.setItem(key, value)
+export const setLocalStorage = (key, value) => {
```
Comment (kellycastrof) [inline src/utils/useLocalStorage.js:1]: Maybe saveToLocalStorage? or addToLocalStorage

File: src/utils/useLocalStorage.js (hunk 2 above)
Comment (dsgarcia8) [reply src/utils/useLocalStorage.js:5]: Agree 👍🏻 , variable name has been changed

File: src/views/Reports/ByMetricReport.jsx [hunk 4]
```diff
@@ -49,9 +50,21 @@ export const ByMetricReport = ({ fromUniverseOverview }) => {
     metricPeersComparison,
     metricIsLoading,
     setMetric
-  } = useMetricReport({ fromUniverseOverview, selectedMetric: 'actuals_revenue' })
+  } = useMetricReport({ fromUniverseOverview, selectedMetric: JSON.parse(localStorage.getItem('metric')) || 'actuals_revenue' })
```
[the remark this answers is not in the document]
Comment (dsgarcia8) [reply src/views/Reports/ByMetricReport.jsx:53]: solved comment

File: src/utils/useLocalStorage.js (hunk 3 above)
Comment (dsgarcia8) [reply src/utils/useLocalStorage.js:1]: 👍🏻

**Review by kellycastrof — APPROVED** [review]

**Pull request merged**